zebra/zapi_msg.c

```diff
@@ -2253,12 +2253,19 @@ static inline void zread_rule(ZAPI_HANDLER_ARGS)
 			}
 		}
 
-		if (!is_default_prefix(&zpr.rule.filter.src_ip))
-			zpr.rule.filter.filter_bm |= PBR_FILTER_SRC_IP;
+		if (ifindex) {
+			if (!is_default_prefix(&zpr.rule.filter.src_ip))
+				zpr.rule.filter.filter_bm |= PBR_FILTER_SRC_IP;
 
-		if (!is_default_prefix(&zpr.rule.filter.dst_ip))
-			zpr.rule.filter.filter_bm |= PBR_FILTER_DST_IP;
+			if (!is_default_prefix(&zpr.rule.filter.dst_ip))
+				zpr.rule.filter.filter_bm |= PBR_FILTER_DST_IP;
+		} else {
+			if (zpr.rule.filter.src_ip.family == AF_INET)
```

Comment:
So PBR_FILTER_SRC_IP is being used as both a v4 indicator as well as a that we should signal on SRC.  This is confusing and frankly breaks v6 for this section of code.  1) The code should be v4/v6 agnostic.  2) it should be an illegal condition that v4 and v6 families are passed down from an upper level protocol

Reply:
same remark. code will not be modified. not necessary actually.